### PR TITLE
Updated get_registrations and post_attendee tests

### PR DESCRIPTION
**Pull request by dunkOnIT**
Outstanding items: 
- [x] Swaggerize output and check Swagger UI (help required on accessing the UI) 
- [x] Discuss merging tests with a FAILING flag
- [ ] Review diff

### CODE REVIEW

**Comment by dunkOnIT** [comment]
- [ ] Move REGISTRATION_STATUS to model
- [ ] Move competition errors to registration errors

Research: 
- [ ] Research filter/map functions

**Comment by dunkOnIT** [comment]
- [ ] Get all tests passing with current code state
- [ ] Set up a pre-commit hook to run rubocop
- [ ] Run rubocop
- [ ] Remove allow_net_connect and check if tests run

File: app/helpers/competition_api.rb [hunk 1]
```diff
@@ -31,9 +50,15 @@ def self.competition_open?(competition_id)
   end
 
   def self.competition_exists?(competition_id)
-    Rails.cache.fetch(competition_id, expires_in: 5.minutes) do
+    competition_info = Rails.cache.fetch(competition_id, expires_in: 5.minutes) do
       self.fetch_competition(competition_id)
     end
+
+    if competition_info[:error] == false
```
Comment (FinnIckler) [inline app/helpers/competition_api.rb:57]: `!competition_info[:error] == false`

File: app/helpers/competition_api.rb [hunk 2]
```diff
@@ -23,6 +23,25 @@ def self.fetch_competition(competition_id)
     end
   end
 
+  def self.get_competition_info(competition_id)
+    competition_info = Rails.cache.fetch(competition_id, expires_in: 5.minutes) do
+      self.fetch_competition(competition_id)
+    end
+
+    if competition_info[:error] == false
```
Comment (FinnIckler) [inline app/helpers/competition_api.rb:31]: `!competition_info[:error] == false`

File: app/helpers/competition_api.rb [hunk 3]
```diff
@@ -23,6 +23,25 @@ def self.fetch_competition(competition_id)
...
+      competition_info[:competition_exists?] = true
```
Comment (FinnIckler) [inline app/helpers/competition_api.rb:32]: reconsider if we need these two booleans

**Review by FinnIckler — CHANGES_REQUESTED** [review]

File: app/helpers/competition_api.rb (hunk 3 above)
Comment (dunkOnIT) [reply app/helpers/competition_api.rb]: I think it makes sense to leave them in and then refactor the competition_api.rb once the PR is merged, as discussed.

File: app/controllers/registration_controller.rb [hunk 4]
```diff
@@ -26,37 +28,69 @@ class RegistrationController < ApplicationController
...
+    puts @user_id
     @competition_id = registration_params[:competition_id]
+    puts @competition_id
     @event_ids = registration_params[:competing]["event_ids"]
+    puts @event_ids
     status = ""
     cannot_register_reason = nil
 
-    unless @current_user == @user_id.to_s
+    # This could be split out into a "validate competition exists" method
```
[the remark this answers is not in the document]
Comment (dunkOnIT) [reply app/controllers/registration_controller.rb:34]: Will do in a subsequent PR, created #245 to keep track of it

**Comment by dunkOnIT** [comment]
@FinnIckler ready for final review!

**Review by FinnIckler — APPROVED** [review]

**Pull request merged**